axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/AbstractFixedAssetLineComputationServiceImpl.java

```diff
@@ -251,36 +262,54 @@ public void multiplyLinesBy(List<FixedAssetLine> fixedAssetLineList, BigDecimal
 
   protected BigDecimal computeInitialDepreciation(FixedAsset fixedAsset, BigDecimal baseValue)
       throws AxelorException {
-    // Theses cases is for when user want to depreciate in one year.
-    // This case is if list is not empty when calling this method
-    if (getFixedAssetLineList(fixedAsset) != null
-        && numberOfDepreciationDone(fixedAsset)
-            .equals(getNumberOfDepreciation(fixedAsset).subtract(BigDecimal.ONE))
-        && !fixedAsset.getFixedAssetCategory().getIsProrataTemporis()) {
-      if (fixedAssetFailOverControlService.isFailOver(fixedAsset)
-          && getComputationMethodSelect(fixedAsset)
-              .equals(FixedAssetRepository.COMPUTATION_METHOD_LINEAR)) {
-        return fixedAssetLineToolService.getCompanyScaledValue(
-            baseValue, getAlreadyDepreciatedAmount(fixedAsset), fixedAsset, BigDecimal::subtract);
+
+    boolean hasLines = getFixedAssetLineList(fixedAsset) != null;
+    boolean isProrata = fixedAsset.getFixedAssetCategory().getIsProrataTemporis();
```

Comment:
Can you check that the fixed asset is not null and the category too to avoid NPE